### PR TITLE
Use personal env file for local development without breaking the CI

### PR DESCRIPTION
**Pull request by danduk82**
So that we can use a personal env file, that is not committed

### CODE REVIEW

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/+dot+gitignore_tmpl [hunk 1]
```diff
@@ -7,6 +7,7 @@ __pycache__/
 /.upgrade.yaml
 /.UPGRADE*
 /geoportal/{{package}}_geoportal/locale/{{package}}_geoportal-client.pot
+env.personal
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/+dot+gitignore_tmpl:10]:
```suggestion
/env.personal
```

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/project.yaml_tmpl [hunk 2]
```diff
@@ -14,5 +14,6 @@ env:
   files:
     - env.default
     - env.project
+    - env.personal
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/project.yaml_tmpl:17]: not needed

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl [hunk 3]
```diff
@@ -145,8 +147,14 @@ def main():
...
-            with open(file_) as src:
-                dest.write(src.read() + "\n")
+            try:
+                with open(file_) as src:
+                    dest.write(src.read() + "\n")
+            except FileNotFoundError as e:
+                if file_ == 'env.personal':
+                    pass
+                else:
+                    raise e
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl:157]: not needed

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl [hunk 4]
```diff
@@ -33,6 +33,8 @@ def load_env(env_files):
     """
     env = {}
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl:34]:
```
if  os.path.exists('env.personal'):
    env_files.append('env.personal')
```

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl [hunk 5]
```diff
@@ -33,6 +33,8 @@ def load_env(env_files):
...
     for env_file in env_files:
+        if not os.path.exists(env_file) and env_file == 'env.personal' :
+            continue
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl:37]: not needed

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl [hunk 6]
```diff
@@ -97,7 +99,7 @@ def main():
     env_files = [e.format(*args.env) for e in project_env["files"]]
     print("Use env files: {}".format(", ".join(env_files)))
     for env_file in env_files:
-        if not os.path.exists(env_file):
+        if not os.path.exists(env_file) and env_file != 'env.personal' :
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl:102]: not needed

**Review by sbrunner — COMMENTED** [review]
Purpose a real simpler solution, documentation souldn't be updated?

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl (hunk 4 above)
Comment (danduk82) [reply geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl:34]: joli

**Comment by ochriste** [comment]
why do we need something specific like that? 
is it just because you dont want to commit into the project your own build config ?

**Review by sbrunner — APPROVED** [review]

**Comment by danduk82** [comment]
@ochriste to avoid pushing to github the local config for development, therefore avoiding to have all the personal stuff of every developer in the project. Also, you might not want to put some credentials in committed files and use this one instead

**Comment by ochriste** [comment]
mmm, I can understand the credential part, but that a rare case.
Personally I will continue commiting into project my build config, I have other things to worry about than trying to recover my build config from somewhere else

**Comment by danduk82** [comment]
it was mainly for the creds indeed

**Pull request merged**